scripts/ci/check-dependencies-versions-match.js

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+// This script ensures that all "dependencies" in package JSONs listed below, use the same versions of
+// dependencies. It also checks that all versions and are pinned, and they don't use the caret operator "^".
+// If you provide "--fix" argument, the script will automatically fix the errors for you.
+
+const semver = require( 'semver' );
+const { globSync } = require( 'glob' );
+const fs = require( 'fs-extra' );
+const upath = require( 'upath' );
+
+const shouldFix = process.argv[ 2 ] === '--fix';
+const [ packageJsons, pathMappings ] = getPackageJsons( [
+	'package.json',
+	'packages/*/package.json',
+	'external/ckeditor5-internal/packages/*/package.json',
+	'external/ckeditor5-internal/package.json',
+	'external/collaboration-features/packages/*/package.json',
+	'external/collaboration-features/package.json'
+] );
+
+const expectedDependencies = getExpectedDepsVersions( packageJsons );
+
+if ( shouldFix ) {
+	fixDependenciesVersions( expectedDependencies, packageJsons, pathMappings );
+} else {
+	checkDependenciesMatch( expectedDependencies, packageJsons );
+}
+
+/**
+ * @param {Object.<String, String>} expectedDependencies
+ * @param {Array.<Object>} packageJsons
+ * @param {Object.<String, String>} pathMappings
+ */
+function fixDependenciesVersions( expectedDependencies, packageJsons, pathMappings ) {
+	packageJsons
+		.filter( packageJson => packageJson.dependencies )
+		.forEach( packageJson => {
+			Object.entries( packageJson.dependencies )
+				.forEach( ( [ dependency, version ] ) => {
+					if ( version !== expectedDependencies[ dependency ] ) {
+						packageJson.dependencies[ dependency ] = expectedDependencies[ dependency ];
+					}
+				} );
+
+			fs.writeJsonSync( pathMappings[ packageJson.name ], packageJson, { spaces: 2 } );
+		} );
+
+	console.log( '✅ All dependencies fixed!' );
+}
+
+/**
+ * @param {Object.<String, String>} expectedDependencies
+ * @param {Array.<Object>} packageJsons
+ */
+function checkDependenciesMatch( expectedDependencies, packageJsons ) {
+	const errors = packageJsons
+		.filter( packageJson => packageJson.dependencies )
+		.flatMap( packageJson => Object.entries( packageJson.dependencies )
+			.map( ( [ dependency, version ] ) => {
+				if ( version !== expectedDependencies[ dependency ] ) {
+					return getWrongVersionErrorMsg( dependency, packageJson.name, version, expectedDependencies );
+				}
+			} )
+			.filter( Boolean )
+		);
+
+	if ( errors.length ) {
+		console.error( '❌ Errors found. Run this script with `--fix` to resolve the issues automatically.' );
+		console.error( errors );
```

Comment:
In case of errors, it prints the array. It would look better with just strings:

![errors](https://github.com/ckeditor/ckeditor5/assets/56868128/c8bbe5f7-5d2e-4ad7-9824-96ece726098e)

If there are no errors, the scripts just ends without any info that everything is fine.